From effa59d218f4766d3f51d216305c02cc8e7aeec6 Mon Sep 17 00:00:00 2001
From: Jordan Johnston <johnstonljordan@gmail.com>
Date: Wed, 25 May 2022 08:10:05 -0500
Subject: [PATCH] mm: Increase max map count

 - this is a bit nuts, but could avoid some potential crashes in Wine.

---
 include/linux/mm.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/linux/mm.h b/include/linux/mm.h
index 5744a3fc4716..0142b447e19f 100644
--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -195,8 +195,7 @@ static inline void __mm_zero_struct_page(struct page *page)
  * not a hard limit any more. Although some userspace tools can be surprised by
  * that.
  */
-#define MAPCOUNT_ELF_CORE_MARGIN	(5)
-#define DEFAULT_MAX_MAP_COUNT	(USHRT_MAX - MAPCOUNT_ELF_CORE_MARGIN)
+#define DEFAULT_MAX_MAP_COUNT	(16777216)
 
 extern int sysctl_max_map_count;
 
-- 
2.36.1

